### PR TITLE
Add unit testing in front end source code

### LINKED ISSUE #2193: Add tests to React submission all section with filtering
See https://github.com/OpenTechFund/hypha/pull/2184

### PR DESCRIPTION
**Pull request by vimal1083**
Fixes #2193

Have written functional unit test cases for 53% of frontend code base. Also added a step  in circle CI to test frontend codebase.

Screen Shot 2021-02-05 at 21 26 46

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+                        presets: ['@babel/preset-react',
```
Comment (Parbhat) [inline .babelrc:2]: Improper indentation

**Review by Parbhat — APPROVED** [review]
Looks good. 1 minor update.

**Review by frjo — APPROVED** [review]
Looks good. Nice to have test on this part of the app as well. Seems to run quite fast as well.

File: .circleci/config.yml [hunk 2]
```diff
@@ -99,6 +100,24 @@ jobs:
...
+          command: sudo chown -R circleci:circleci /usr/local
+
+      - run:
+          name: Run functional unit test cases [REACT.JS]
+          command: |
+            npm install --quiet
+            npm install -g gulp-cli
+            npm run test
+
   test:
```
[the remark this answers is not in the document]
Comment (vimal1083) [reply .circleci/config.yml]: Changed it. Can you update the branch protection rules and make `test-be` and `test-fe` mandatory?

File: .babelrc (hunk 1 above)
Comment (vimal1083) [reply .babelrc]: done

**Comment by Parbhat** [comment]
@frjo This PR is ready to merge.

**Pull request merged**